modules/sdk-core/src/bitgo/utils/tss/eddsa/eddsa.ts

```diff
@@ -32,6 +34,50 @@ import { KeychainsTriplet } from '../../../baseCoin';
  */
 
 export class EddsaUtils extends baseTSSUtils<KeyShare> {
+  async verifyWalletSignatures(
+    userGpgPub: string,
+    backupGpgPub: string,
+    bitgoKeychain: Keychain,
+    decryptedShare: string,
+    verifierIndex: 1 | 2
+  ): Promise<void> {
+    assert(bitgoKeychain.commonKeychain);
+    assert(bitgoKeychain.walletHSMGPGPublicKeySigs);
+
+    const bitgoGpgKey = await getBitgoGpgPubKey(this.bitgo);
+
+    const userKeyPub = await openpgp.readKey({ armoredKey: userGpgPub });
+    const userKeyId = userKeyPub.keyPacket.getFingerprint();
+
+    const backupKeyPub = await openpgp.readKey({ armoredKey: backupGpgPub });
+    const backupKeyId = backupKeyPub.keyPacket.getFingerprint();
+
+    const walletSignatures = await openpgp.readKeys({ armoredKeys: bitgoKeychain.walletHSMGPGPublicKeySigs });
+    if (walletSignatures.length !== 2) {
+      throw new Error('Invalid wallet signatures');
+    }
```

Comment:
https://github.com/BitGo/BitGoJS/pull/2920/files#r1025889093

@Logicwax so we probably want to add a check here fir walletSignatures[0] and walletSignatures[1] given the ^ discussion.

Reply:
yep

Reply:
Just pushed some new code with the requested checks